src/components/shared/ErrorOverlay.tsx

```
@@ -0,0 +1,531 @@
+  }
+
+  // Categorize error
+  const category = categorizeError(error);
+  const displayTitle = title ?? getErrorTitle(category);
+  const suggestion = getRecoverySuggestion(category);
+
+  // Critical errors cannot be dismissed
+  const canDismiss = severity !== 'critical' && onDismiss;
+
+  // Handle escape key
+  useEffect(() => {
```

Biome (2.3.13)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)